### ISSUE TITLE
Export a table as CSV, fails if more that 512 records are in the table.

### ISSUE THREAD

**Issue by Latorrer** [sqlitebrowser/sqlitebrowser#834]
### Useful extra information

Program writes unreadable characters if I try to export 513 records or more.

#### I'm opening this issue because:

- [x] DB4S has a bug

#### I'm using DB4S on:

- [x] Windows 7

#### I'm using DB4S version:

- [x] 3.9.1

#### I have also:

- [x] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows

**Comment by chrisjlocke** [comment]
Any chance of a copy of the database?  I've a database of 500,000+ rows, and recently had to export, and didn't have any issues...  
I can test with a test database, but if you're having issues, it would be handy to use the same database you're using, in case its something specific with the data.  You can email it to me, and I can confirm the database will be deleted after testing.

**Comment by chrisjlocke** [comment]
A quick export of 229,000+ rows worked fine.

![image](https://cloud.githubusercontent.com/assets/7872218/19466458/ed08b89a-9502-11e6-8844-b6139dbe83bb.png)

**Comment by chrisjlocke** [comment]
And exporting a table (as opposed to exporting a database)

![image](https://cloud.githubusercontent.com/assets/7872218/19467598/ea7b10e4-9509-11e6-829c-6101521152f7.png)

Noticed there was an unnecessary blank line at the end, but certainly not 'unreadable'.

**Comment by Latorrer** [comment]
Chris, have a db I can send, what email address?  Really appreciate the
help!!

**Comment by chrisjlocke** [comment]
whyaintthisworking@chrisjlocke.co.uk is great.  If you can send over the results you got too, that would be handy.... the CSV with goo in it.

**Comment by chrisjlocke** [comment]
CSV file received.  Issue 'solved'.

The CSV file has no carriage returns in it, so all the fields are on one line.
When you export, try setting the 'new line characters' to the Windows option.  I assume this setting is currently blank?

![image](https://cloud.githubusercontent.com/assets/7872218/19492198/a4cf9952-956d-11e6-821a-272275f1dd9f.png)

If no option is set to the quote character, then a NUL character is used.  This is the 'unreadable' characters you see.  

![image](https://cloud.githubusercontent.com/assets/7872218/19492420/8019cfc8-956e-11e6-864d-cdcbc05ad1d6.png)

So in summary, use these options.

![image](https://cloud.githubusercontent.com/assets/7872218/19492441/942b34de-956e-11e6-8da2-3b95646b38f6.png)

A quote character as a .... well, quote character, and CRLF (carriage returns and line feeds) for new line characters.  It'll work wonders! 😄

**Comment by justinclift** [comment]
Ahhh, this makes the third bug report for that same problem then.

@chrisjlocke Do you reckon we should change the default on Windows? (it's pretty easy to do)

**Comment by chrisjlocke** [comment]
Yes, I would suggest so.  It appears this option is getting missed.  I would change the quote option (NUL characters isn't helpful - quote characters are more 'traditional') and the newline characters.  Added the 'enhancement' label...

**Comment by justinclift** [comment]
@Latorrer have you had a moment to verify the approach suggested by @chrisjlocke works?

**Comment by Latorrer** [comment]
It does!!! Thanks again Chris!

**Comment by justinclift** [comment]
Thanks @Latorrer. :smile:

Good stuff @chrisjlocke, well done. 😀
